app/models/manageiq/providers/azure/cloud_manager/vm_or_template_shared.rb

```diff
@@ -37,17 +42,53 @@ def vm_object
     @vm_object ||= provider_service.get(name, resource_group.name)
   end
 
+  def image_object
+    @image_object ||= image_service.get(name, managed_resource_group)
+  end
+
   def os_disk
-    @os_disk ||= vm_object.properties.storage_profile.os_disk
+    @os_disk ||= if template
+                   image_object.properties.storage_profile.os_disk
+                 else
+                   vm_object.properties.storage_profile.os_disk
+                 end
   end
 
   delegate :managed_disk?, to: :vm_object
 
+  def managed_image?
+    return false unless ems_ref =~ /^\/subscriptions\//i
+    os_disk.try(:managed_disk) ? true : false
+  end
+
+  def managed_resource_group
+    return nil unless ems_ref =~ /^\/subscriptions\//i
+    ref_parts = ems_ref.split('/')
+    if ref_parts[3] =~ /resourceGroups/i
+      return ref_parts[4]
```

Comment:
There should be a resource_group association on the Rails side for current code.

Reply:
@djberg96 shoulda woulda coulda... :-)   If you recall I asked you about this because there wasn't a resource group.  Its nil.  And needs to work in Fine too, where there is no association.

Reply:
Hmmm...